packages/ui/src/components/TextArea/index.tsx

```diff
@@ -0,0 +1,288 @@
+import type { DOMAttributes, ReactNode } from 'react'
+import { forwardRef, useId } from 'react'
+import { Button, SIZE_HEIGHT as ButtonSizeHeight } from '../Button'
+import { Row } from '../Row'
+import { Stack } from '../Stack'
+import { Text } from '../Text'
+import { Tooltip } from '../Tooltip'
+
+const STATE_ICON_SIZE = 16
+
+const StyledTextAreaWrapper = styled.div`
+  position: relative;
+`
+
+const StyledTextAreaAbsoluteStack = styled(Stack)`
+  position: absolute;
+  top: ${({ theme }) => theme.space['2']};
+  right: ${({ theme }) => theme.space['1']};
+`
+
+type StyledTextAreaProps = {
+  isSuccess: boolean
+  isError: boolean
+  isClearable: boolean
+}
+const StyledTextArea = styled('textarea', {
+  shouldForwardProp: prop =>
+    !['isSuccess', 'isError', 'isClearable'].includes(prop),
+})<StyledTextAreaProps>`
+  width: 100%;
+  resize: vertical;
+  background: ${({ theme }) => theme.colors.neutral.background};
+  border: 1px solid ${({ theme }) => theme.colors.neutral.border};
+  color: ${({ theme }) => theme.colors.neutral.text};
+  &::placeholder {
+    color: ${({ theme }) => theme.colors.neutral.textWeak};
+  }
+  border-radius: ${({ theme }) => theme.radii.default};
+  padding: ${({ theme }) =>
+    `${theme.space['1.5']} ${theme.space['1']} ${theme.space['1.5']} ${theme.space['2']}`};
+  padding-right: ${({ theme, isClearable, isError, isSuccess }) =>
+    /* including 1 optional if both element is visible + 1 because content is absolute 1space unit from right */
+    `calc(${theme.space[isClearable && (isSuccess || isError) ? '4' : '3']} + ${
+      isClearable ? `${ButtonSizeHeight.xsmall}px` : '0px'
+    } + ${isSuccess || isError ? `${STATE_ICON_SIZE}px` : '0px'})`};
+
+  ${({ theme, disabled }) =>
+    !disabled
+      ? `
+  &:hover {
+    border-color: ${theme.colors.primary.border};
+  }
+  &:focus {
+    outline: none;
+    border-color: ${theme.colors.primary.border};
+    box-shadow: ${theme.shadows.focusPrimary};
+  }
+  `
+      : ''}
+
+  ${({ theme, isSuccess, isError, readOnly, disabled }) => {
+    if (disabled) {
+      return `
+            background : ${theme.colors.neutral.backgroundDisabled};
+            border-color : ${theme.colors.neutral.borderDisabled};
+            color : ${theme.colors.neutral.textDisabled};
+            &::placeholder {
+              color: ${theme.colors.neutral.textWeakDisabled};
+            }
+        `
+    }
+    if (readOnly) {
+      return `
+            background : ${theme.colors.neutral.backgroundWeak};
+            border-color : ${theme.colors.neutral.borderWeak};
+        `
+    }
+    if (isSuccess) {
+      return `
+            border-color : ${theme.colors.success.border};
+        `
+    }
+    if (isError) {
+      return `
+            border-color : ${theme.colors.danger.border};
+        `
+    }
+
+    return ''
+  }}
+`
+
+type TextAreaProps = {
+  id?: string
+  autoFocus?: boolean
+  label: string
+  value: string
```

Comment:
Why value is required? If it is required the placeholder has no utility

Reply:
placeholder is displayed if value is an empty string, however form might provide undefined indeed